src/scales/scale.linearbase.js

```diff
@@ -155,14 +172,15 @@ module.exports = function(Chart) {
 			var tickOpts = opts.ticks;
 
 			// Figure out what the max number of ticks we can support it is based on the size of
-			// the axis area. For now, we say that the minimum tick spacing in pixels must be 50
+			// the axis area. For now, we say that the minimum tick spacing in pixels must be 40
 			// We also limit the maximum number of ticks to 11 which gives a nice 10 squares on
 			// the graph. Make sure we always have at least 2 ticks
 			var maxTicks = me.getTickLimit();
 			maxTicks = Math.max(2, maxTicks);
 
 			var numericGeneratorOptions = {
 				maxTicks: maxTicks,
+				maxTicksLimit: tickOpts.maxTicksLimit,
```

Comment:
`maxTicks` and `maxTicksLimit` are quite confusing (naming but also intent). `maxTicks` depends on `maxTicksLimit` in  `getTickLimit()` and since this is exposed through the public API, I think we should try to keep only `maxTicks` (note that in `generateTicks()`, you only use one of them).

Do we really need to apply the following logic:
> If `stepSize` is set but `maxTicksLimit` is not, spacing is set to `stepSize`.

... or maybe we should express `maxTicks` based on `stepSize` in `getTickLimit` instead of returning `11`,

... or maybe `numericGeneratorOptions.maxTicks` should be undefined if `stepSize && !maxTicksLimit`.

Anything that would avoid exposing `maxTicksLimit` in the generator options and prevent special cases would be great.